data/data-pipeline/data_pipeline/comparison_tool/src/tract_comparison__template.ipynb

```diff
@@ -0,0 +1,379 @@
+    "plt.savefig(os.path.join(OUTPUT_DATA_PATH, \"tract_lvl_avg.jpg\"), bbox_inches='tight')"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "713f2a04-1b2a-472c-b036-454cd5c9a495",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "#papermill_description=Tract_stats_grouped\n",
+    "tract_level_by_grouping = utils.get_tract_level_grouping(\n",
+    "    joined_frame,\n",
+    "    SCORE_COLUMN,\n",
+    "    COMPARATOR_COLUMN,\n",
+    "    ADDITIONAL_DEMO_COLUMNS + DEMOGRAPHIC_COLUMNS,\n",
+    ")\n",
+    "\n",
+    "tract_level_by_grouping_formatted = utils.format_multi_index_for_excel(\n",
+    "    tract_level_by_grouping\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "0d166f53-4242-46f1-aedf-51dac517de94",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "tract_level_by_grouping_formatted"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "372ac341-b03b-404a-9462-5087b88579b7",
+   "metadata": {},
+   "source": [
+    "## Population-weighted stats"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "d172e3c8-60ab-44f0-8034-dda69f1146da",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "#papermill_description=Population_stats\n",
+    "population_weighted_stats = pd.concat(\n",
+    "    [\n",
+    "        utils.construct_weighted_statistics(\n",
+    "            joined_frame,\n",
+    "            COMPARATOR_COLUMN,\n",
+    "            DEMOGRAPHIC_COLUMNS + ADDITIONAL_DEMO_COLUMNS,\n",
+    "            TOTAL_POPULATION_COLUMN,\n",
+    "        ),\n",
+    "        utils.construct_weighted_statistics(\n",
+    "            joined_frame,\n",
+    "            SCORE_COLUMN,\n",
+    "            DEMOGRAPHIC_COLUMNS + ADDITIONAL_DEMO_COLUMNS,\n",
+    "            TOTAL_POPULATION_COLUMN,\n",
+    "        ),\n",
+    "    ],\n",
+    "    axis=1,\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "751d2d21",
+   "metadata": {},
+   "source": [
+    "## Final information about overlap"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "9eef9447",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "also_cejst, states = utils.get_final_summary_info(\n",
+    "    population=population, comparator_file=COMPARATOR_FILE, geoid_col=GEOID_COLUMN\n",
+    ")\n",
+    "display(also_cejst)\n",
+    "states_text = \"States included in comparator: \" + states\n",
+    "states_text"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "a9fb4dfa-876d-4f25-9552-9831f51ec05c",
+   "metadata": {},
+   "source": [
+    "## Print to excel"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "67e2c71e-68af-44b4-bab0-33bc54afe7a4",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "#papermill_description=Writing_excel\n",
+    "utils.write_single_comparison_excel(\n",
+    "    OUTPUT_EXCEL,\n",
```

Comment:
great example where named arguments could prevent a lot of accidental off-by-one argument list errors!